apps/openchallenges/infra/cdk/README.md

```diff
@@ -2,51 +2,268 @@
 
 AWS CDK infrastructure as code for OpenChallenges. This project defines and manages the cloud resources required to deploy and operate the OpenChallenges application stack.
 
+## Project Structure
+
+```
+openchallenges-infra-cdk/
+├── openchallenges_infra_cdk/
+│   ├── shared/              # Shared libraries used by all environments
+│   │   ├── config.py        # Configuration management
+│   │   ├── naming.py        # Resource naming utilities
+│   │   ├── constructs/      # Reusable CDK constructs
+│   │   └── stacks/          # Reusable CDK stacks
+│   ├── dev/                 # Development environment app
+│   │   └── app.py
+│   ├── stage/               # Staging environment app
+│   │   └── app.py
+│   └── prod/                # Production environment app
+│       └── app.py
+├── tests/
+│   ├── unit/                # Unit tests
+│   └── integration/         # Integration tests
+├── .env                     # Base environment variables
+├── .env.dev                 # Development-specific variables
+├── .env.stage               # Staging-specific variables
+├── .env.prod                # Production-specific variables
+└── cdk.json                 # CDK configuration
+```
+
 ## Quick Start
 
+### Prerequisites
+
+- AWS CLI configured with appropriate credentials
+- Python 3.11 or later
+- Node.js and pnpm (for Nx commands)
+
+### Environment Configuration
+
+The project uses environment variables for configuration. Nx automatically loads variables from `.env` files based on the target configuration.
+
+#### Development Environment
+
+For the development environment, you must set `DEVELOPER_NAME` to create unique resource names:
+
+```bash
+# .env.dev
+DEVELOPER_NAME=jsmith
```

Comment:
I'm a fan of passing in variables using json or yaml so that I can directly pass in more complex parameters like lists and dictionaries.   I believe that is also a built in functionality of the CDK.   Any thought of going with that approach?

Reply:
Yes, the best approach would be a combination of config file (YAML or JSON) + environment variables. I will look for the recommended why to achieve this for CDK as the number of config parameters grows.